core/src/checks/rpc_checks.c

```diff
@@ -0,0 +1,202 @@
+#include "checks.h"
+#include "config.h"
+#include "log.h"
+#include "nanopb_stream.h"
+#include "print.h"
+#include "rand.h"
+#include "rpc.h"
+
+#include <assert.h>
+#include <pb_decode.h>
+#include <pb_encode.h>
+#include <squareup/subzero/internal.pb.h>
+
+static size_t get_serialized_message_size(const InternalCommandRequest* const cmd) {
+  pb_ostream_t stream = PB_OSTREAM_SIZING;
+  if (!pb_encode_delimited(&stream, InternalCommandRequest_fields, cmd)) {
+    ERROR("%s: pb_encode_delimited() failed: %s", __func__, PB_GET_ERROR(&stream));
+    return 0;
+  }
+  return stream.bytes_written;
+}
+
```

Comment:
nit: future me will probably struggle a bit to debug a test failure here if it happens? would you be opposed to breaking DRY a bit to add more comment bread crumbs to state the intended narrative of what the test is doing?